docs/rfc/column-level-lineage.md

```diff
@@ -0,0 +1,726 @@
+    List<DerivedBy> list = new ArrayList();
+
+    for(FineGrainUpstream stream: upstreamLineage.getUpstreams()) {
+      for(FineGrainFieldUpstream fieldUpStream: stream.getFields()) {
+        try {
+          list.add(new DerivedBy()
+              .setSource(Urn.createFromString(stream.getDataset().toString() + ":" + fieldUpStream.getSourceField()))
+              .setDestination(Urn.createFromString(urn.toString() + ":" + fieldUpStream.getTargetField())));
+        } catch (URISyntaxException e) {
+
+        }
+      }
+    }
+
+    return Collections.singletonList(new GraphBuilder.RelationshipUpdates(list, REMOVE_ALL_EDGES_FROM_SOURCE));
+  }
+
+}
+
+```
+
+
+> pay attention on how the `source urn is set` in `.setSource(Urn.createFromString(stream.getDataset().toString() + ":" + fieldUpStream.getSourceField()))`
+and `destination urn is set` in `setDestination(Urn.createFromString(urn.toString() + ":" + fieldUpStream.getTargetField())`
+
+The base `BaseRelationshipBuilder` defines there is an urn always pointing to the dataset having this [FineGrainUpstreamLineage] relationship aspect.  And this `urn` is the destination of upstream lineage.
+
+The final step, we register these two relationship builders with `DatasetGraphBuilder`
+```
+Collections.unmodifiableSet(new HashSet<BaseRelationshipBuilder>() {
+        {
+          add(new DownstreamOfBuilderFromUpstreamLineage());
+          add(new HasFieldBuilderFromSchemaMetadata());
+          add(new OwnedByBuilderFromOwnership());
+          add(new DerivedByBuilderFromFineGrainUpstreamLineage());
+        }
+      });
+```
+
+
+
+
+## How we teach this
+This implementation supports features 
+1. relationship between a dataset and its fields
+2. relationship between a dataset and the other dataset (already available)
+3. relationship between a dataset field derived by **one or more fields**. 
+
+In this Neo4j graph presentation, 
+1. dataset `c_bar_13` has fields of `bar131`, `bar132`, `bar133` and `col134`;
+2. `p_foo_13` dataset has fields of foo131, foo132, foo133, and foo134
+3. 'p_2_foo_13` dataset has fields of 2_foo131, 2_foo132, 2_foo133, and 2_foo134
+4. dataset `c_bar_13` is downstream of `p_foo_13` and `p_2_foo_13`
+5. bar131 of c_bar_13 is derived by foo131 and 2_foo131
+6. bar 132 of c_bar_13 is derived by foo132 and 2_foo132
+
+
+
+![Screen Shot 2020-08-04 at 11 42 54 AM](https://user-images.githubusercontent.com/356347/89320841-a8215400-d647-11ea-882b-8c9dea4bdbf6.png)
+
+
+
+## Drawbacks
+Because `SchemaField` is not a first-class entity in Datahub, the current [Neo4jGraphWriterDao](https://github.com/linkedin/datahub/blob/master/metadata-dao-impl/neo4j-dao/src/main/java/com/linkedin/metadata/dao/internal/Neo4jGraphWriterDAO.java) only supports the first class entity by enforcing [validation rule] implemented by [RelationshipValidator](https://github.com/linkedin/datahub/blob/master/metadata-validators/src/main/java/com/linkedin/metadata/validator/RelationshipValidator.java)
```

Comment:
Thanks for your work with work arounds to get this working. 

Search and Discovery of fields and its associated metadata is also one of the common asks I believe. We might also want to think more inlines of making datasetfield as an entity in itself.